packages/aws-cdk/lib/init-templates/v1/app/csharp/add-project.hook.ts

```diff
@@ -3,10 +3,13 @@ import * as path from 'path';
 import { InvokeHook } from '../../../../init';
 
 export const invoke: InvokeHook = async (targetDirectory: string) => {
+  const escapeSpaces = (value: string) => {
+    return value.replace(' ', '\\ ');
+  };
   const slnPath = path.join(targetDirectory, 'src', '%name.PascalCased%.sln');
   const csprojPath = path.join(targetDirectory, 'src', '%name.PascalCased%', '%name.PascalCased%.csproj');
 
-  const child = child_process.spawn('dotnet', ['sln', slnPath, 'add', csprojPath], {
+  const child = child_process.spawn('dotnet', ['sln', escapeSpaces(slnPath), 'add', escapeSpaces(csprojPath)], {
```

Comment:
I'm surprised that `child_process.spawn` doesn't add quotes around the arguments in the first place. I don't understand what the point of the argument array is without doing that.

Can you find documentation on the internet around this behavior?

Reply:
You might be able to use this function instead: https://github.com/aws/aws-cdk/blob/3d71e44840976f3667af879fde11be0b7ab6f80c/packages/aws-cdk/lib/os.ts#L15